src/router.js

```diff
@@ -69,6 +69,58 @@ router.post('/applications', async (request, response) => {
   }
 });
 
+/**
+ * Clean the incoming POST request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {any} existingId The ID of the license that the sett belongs to.
+ * @param {any} body The incoming request's body.
+ * @returns {any} A json object that's just got our cleaned up fields on it.
+ */
+const cleanSettInput = (existingId, body) => {
+  return {
+    ApplicationId: existingId,
+    // The strings are trimmed for leading and trailing whitespace and then
+    // copied across if they're in the POST body or are set to undefined if
+    // they're missing.
+    sett: body.sett === undefined ? undefined : body.sett.trim(),
+    gridRef: body.gridRef === undefined ? undefined : body.gridRef.trim(),
```

Comment:
This line needs to be `gridRef: body.gridReference === undefined ? undefined : body.gridReference.trim(),` as the field in the body of the request is `gridReference`.